### PR TITLE
Argo docs update

### PR DESCRIPTION
**Pull request by kcpevey**
The Argo docs are lacking in depth and clarity. This PR adds more details on how things work and more examples on how to run a workflow on Argo using python. 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

**Comment by kcpevey** [comment]
This is ready for review. I created an `argo-assistant` package which is in good shape, but isn't fully fleshed out (https://github.com/nebari-dev/nebari-argo-assistant). However, JATIC is moving away from Argo so I have to wrap all this up. I've chosen to avoid mention of the separate package in the docs for now and instead add all the necessary code into the docs. Its not the best option, but at least the docs will explain how to use Argo for realistic usecases.

File: docs/docs/how-tos/using-argo.md [hunk 1]
```diff
@@ -1,48 +1,118 @@
...
 description: Argo workflow management
 ---
 
 # Automate workflows with Argo
 
-Argo Workflows is an open source container-native workflow engine for orchestrating parallel jobs on Kubernetes. Argo
-workflows comes enabled by default with Nebari deployments.
+[Argo Workflows](https://argoproj.github.io/workflows) is an open source container-native
+workflow engine for orchestrating parallel jobs on Kubernetes. In other words,
+Argo helps you run a sequence on tasks or functions without you having to be
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:11]:
```suggestion
Argo helps you run a sequence of tasks or functions without you having to be
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+The following workflow is perhaps the simplest possible workflow. There are
+quite a few bits here so we'll step through it.
+
+### Global Configuration
+
+Nebari sets up several environment variables with tokens, etc. that enable us to
+use Argo more smoothly. However, there are two global configuration settings
+that we'll need to manually add to each workflow script.
+
+```
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:32]:
```suggestion
```python
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+# Argo Workflows Walkthrough
+
+## Introduction
+
+Using a workflow manager can help you automate ETL pipelines, schedule regular
+analysis, or just chain together a sequence of functions. Argo is available on
+Nebari for workflow management. If you haven't already checked out the
+[introductory documentation on Argo](/how-tos/using-argo.md), head over to that
+[doc page](/how-tos/using-argo.md) and take a look!
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:15]:
```suggestion
Nebari for workflow management. If you haven't already, check out the
[introductory documentation on using Argo](/how-tos/using-argo.md).
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+account token by default, we need to tell Argo which user we are. We also need
+to tell Argo to use the Nebari Workflow Controller so that we have access to
+our Nebari file system and conda environments from within the Argo pod
+([more information](/how-tos/using-argo.md#access-your-nebari-environments-and-file-system-while-on-an-argo-pod-beta)).
+
+The workflow labels must be hexadecimal ASCII while the usernames have no such
+constraint so we have a helper function `sanitize_labels` to ensure that our
+label is valid for Argo.
+
+```
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:53]:
```suggestion
```python
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+### Time to Live Strategy
+
+By default, Argo does not destroy servers after completion of a workflow.
+Because this can cause substantial unexpected cloud costs, we _highly_
+recommend _always_ setting the "Time to live strategy", or `TTLStrategy` on
+every workflow.
+
+Note that the Argo UI will only show the workflow details until the
+`TTLStrategy` time has elapsed so make sure you have enough time to evaluate
+logs, etc. before those details are removed.
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:102]:
```suggestion
:::note 
The Argo UI will only show the workflow details until the
`TTLStrategy` time has elapsed so make sure you have enough time to evaluate
logs, etc. before those details are removed.
:::
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+
+```
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:104]:
```suggestion
```python
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+The `namespace` parameter is set to `dev` by default, but Nebari sets it up
+as part of the environment variables so we'll pull it from there. The
+`generate_name` parameter allows us to give our job a prefix and Argo will add
+a suffix to ensure uniqueness. Lastly, we'll give the workflow an `entrypoint`.
+This parameter needs to match with the name of the `Steps` parameter you're
+using (or `DAG`).
+
+_Workflow constructor_
+
+Let's put this all together and have a closer look.
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:138]:
```suggestion
Let's put this all together in a _Workflow constructor_.
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+
+```
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:140]:
```suggestion
```python
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+"cloud.google.com/gke-nodepool".
+
+The `namespace` parameter is set to `dev` by default, but Nebari sets it up
+as part of the environment variables so we'll pull it from there. The
+`generate_name` parameter allows us to give our job a prefix and Argo will add
+a suffix to ensure uniqueness. Lastly, we'll give the workflow an `entrypoint`.
+This parameter needs to match with the name of the `Steps` parameter you're
+using (or `DAG`).
+
+_Workflow constructor_
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:136]:
```suggestion
### Workflow constructor
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+initialized. We'll grab the Nebari Jupyter image. This has the added benefit
+of bringing parity between running a code on your Nebari instance and running
+on Argo.
+
+As you've seen, we're creating quite a bit of peripheral code and we're about
+to add even more. Let's bring some structure in to help us organize things.
+
+### "Argo Assistant" code
+
+For this, we have a little "Argo Assistant" code that will help us out.
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:205]:
```suggestion
### "Argo Assistant" code

As you've seen, we're creating quite a bit of peripheral code and we're about
to add even more. Let's bring some structure in to help us organize things.
For this, we have a little "Argo Assistant" code that will help us out.
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+
+```
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:207]:
```suggestion
```python
```

File: docs/docs/tutorials/argo-workflows-walkthrough.md [hunk 12]
```diff
@@ -0,0 +1,340 @@
...
+                name="step-name",
+                arguments=[Parameter(name="bash_command", value=conda_command)],
+            )
+
+    workflow = w.create()
+    return workflow
+
+```
+
+First, you'll need to create a python script and a conda environment. Then to
```
Comment (pavithraes) [inline docs/docs/tutorials/argo-workflows-walkthrough.md:326]:
```suggestion
Next, you'll need to create a python script and a conda environment. Then to
```

File: docs/docs/how-tos/using-argo.md [hunk 13]
```diff
@@ -1,48 +1,118 @@
...
+present (it will manage the server resources for you). Argo workflows
+comes enabled by default with Nebari deployments.
 
-## Access Argo Server
+Access control for Argo on Nebari is done through Keycloak user groups. All
+users in the `admin` or `developer` groups have access to Argo.
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:16]:
```suggestion
users in the `admin` or `developer` groups have access to Argo.

:::note
Also see the [Set up Argo Workflows documentation](/how-tos/setup-argo).
:::

```

File: docs/docs/how-tos/using-argo.md [hunk 14]
```diff
@@ -1,48 +1,118 @@
...
 
-If Argo Workflows is enabled, users can access argo workflows server at: `your-nebari-domain.com/argo`. Log in via
-Keycloak with your usual credentials.
+## Ways to access the Argo Server
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:18]:
```suggestion
## Access the Argo Server
```

File: docs/docs/how-tos/using-argo.md [hunk 15]
```diff
@@ -1,48 +1,118 @@
...
 
-## Jupyterflow-Override (Beta)
+You can also access the logs for a workflow using
+`argo logs -n workflow_name @latest`.
+
+For more information on Argo workflows via the UI or the CLI, you can visit the
+[Argo docs](https://argoproj.github.io/argo-workflows/workflow-concepts/).
+
+[Hera](https://hera-workflows.readthedocs.io/) is a framework for building and
+submitting Argo workflows in Python. TODO - reference another doc.
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:81]:
```suggestion
submitting Argo workflows in Python. Learn more in the [Argo Workflows walkthrough tutorial](/tutorials/argo-workflows-walkthrough).
```

File: docs/docs/how-tos/using-argo.md [hunk 16]
```diff
@@ -1,48 +1,118 @@
...
+
+## Access your Nebari environments and file system while on an Argo pod (BETA)
+
+Once you move beyond the "Hello World" Argo examples, you may realize that the
+conda environments and the persistent storage you have on Nebari would be
+really useful in your temporary Argo pods. Lucky for you, we've solved that
+problem for you!
+
+Nebari comes with [Nebari Workflow Controller (BETA)](https://github.com/nebari-dev/nebari-workflow-controller)(NWC)
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:90]: What does NWC mean?

File: docs/docs/how-tos/using-argo.md [hunk 17]
```diff
@@ -1,48 +1,118 @@
...
+which transfers the user's environment variables, home and shared directories,
+docker image, and available conda environments to the server where the Workflow
+is running. Users can then run a script that loads and saves from their home
+directory with a particular conda environment.
+
+All of these things are enabled when users add the `jupyterflow-override` label
+to their workflow as in this example using Hera:
+
+```
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:99]:
```suggestion
```python
```

File: docs/docs/how-tos/using-argo.md (hunk 16 above)
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:90]:
```suggestion
Nebari comes with [Nebari Workflow Controller (BETA)](https://github.com/nebari-dev/nebari-workflow-controller), abbreviated as NWC,
```

**Review by pavithraes — APPROVED** [review]
Thank you for such a thorough set of docs, @kcpevey!

Most suggestions are minor, so I'll approve and we can merge when ready. :)

**Comment by kcpevey** [comment]
Thanks for the help @pavithraes !  I've accepted all of your changes.

File: docs/docs/how-tos/using-argo.md [hunk 18]
```diff
@@ -1,48 +1,123 @@
...
+comes enabled by default with Nebari deployments.
 
-## Access Argo Server
+Access control for Argo on Nebari is done through Keycloak user groups. All
+users in the `admin` or `developer` groups have access to Argo.
 
-If Argo Workflows is enabled, users can access argo workflows server at: `your-nebari-domain.com/argo`. Log in via
-Keycloak with your usual credentials.
+:::note
+Also see the [Set up Argo Workflows documentation](/how-tos/setup-argo).
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:19]:
```suggestion
Also see the [Set up Argo Workflows documentation](/docs/how-tos/setup-argo).
```

File: docs/docs/how-tos/using-argo.md [hunk 19]
```diff
@@ -1,48 +1,123 @@
...
+`argo get @latest` to get the latest submitted workflow.
+
+You can also access the logs for a workflow using
+`argo logs -n workflow_name @latest`.
+
+For more information on Argo workflows via the UI or the CLI, you can visit the
+[Argo docs](https://argoproj.github.io/argo-workflows/workflow-concepts/).
+
+[Hera](https://hera-workflows.readthedocs.io/) is a framework for building and
+submitting Argo workflows in Python. Learn more in the [Argo Workflows walkthrough tutorial](/tutorials/argo-workflows-walkthrough).
```
Comment (pavithraes) [inline docs/docs/how-tos/using-argo.md:86]:
```suggestion
submitting Argo workflows in Python. Learn more in the [Argo Workflows walkthrough tutorial](/docs/tutorials/argo-workflows-walkthrough).
```

**Pull request merged**